client/src/components/shared/StarRating.jsx

```diff
@@ -96,10 +83,10 @@ class StarRating extends React.Component {
     return (
       <div class="star-rating" onMouseOut={mouseOut}>
         {
-          starsFull.map(star =>
```

Comment:
Why separate getters and `.map` methods for full and empty stars? Is there a reason that starsFull and starsEmpty contain integers? Why not just have a single `stars` array of `src` values? Also, this belongs to a prior PR, but we should avoid assigning event listeners to elements that aren't clickable.

Reply:
What do you mean by 'contain integers'? 

Assuming I understand your comment: basically, what I am doing is rendering a set  of full stars, a partial star (if exists), and then a set of empty stars if applicable. With these 3 blocks, the full & empty stars are hard-coded. 

- The full stars are determined by just rounding the stars rating down (floor). 
- The partial star is more complex and is determined by rounding to the nearest increment of 25 (12.5-tiebreaker is rounded up) and then selecting the appropriately-filled star for that (from empty to full). 
- The empty stars are rendered for whatever stars remain between the max stars and the partial star.

Do you have an idea for how to handle this simpler?

I mean, maybe with SVG full you could just render all empty stars as a mask and fill the background to the appropriate proportional width (if that is possible with SVG?).

Part of the implementation was also to accommodate both showing average ratings, and allowing selecting a new rating as separate and simultaneous capabilities of the component, which brings us to your last comment:

The elements ARE clickable, as this component handles both rendering average ratings as well as submitting new ratings. I handled this by determining clickability based on whether or not the user supplies a callback function (which transmits the clicked state). However, you can't add attributes programmatically as far as I can tell, so rather than rendering two sets of stars that are identical except for the presence of the onClick attribute, I chose instead to either have onClick attached to the callback, or to a callback that does nothing. CSS handles the cursor state for indicating to the user if it is 'clickable'.

What are your thoughts on this or a better way to handle this? Perhaps more easily discussed in Zoom?

Reply:
Beautiful. This is better than what I had in mind. Thank you.